system/HTTP/ResponseTrait.php

```diff
@@ -566,7 +566,7 @@ public function getCookieStore()
      */
     public function hasCookie(string $name, ?string $value = null, string $prefix = ''): bool
     {
-        $prefix = $prefix ?: Cookie::setDefaults()['prefix']; // to retain BC
+        $prefix = $prefix !== '' && $prefix !== '0' ? $prefix : Cookie::setDefaults()['prefix']; // to retain BC
```

Comment:
i think the prefiix checks are against empty strings only, so no need for falsy string checks?